### PR TITLE
implement cos gradient as a function op

### PR DESCRIPTION
**Pull request by prathikr**
### Description
Implemented gradient of cos as per the function below.
image

### Motivation and Context
Cos gradient required for [huggingface's diffusers library](https://github.com/huggingface/diffusers)

### Testing
built ORT from source: `./build.sh --config RelWithDebInfo --enable_training --use_cuda --cuda_home /usr/local/cuda --cudnn_home /usr/local/cuda --build_wheel --parallel --skip_tests`
tested CosGrad implementation: `cd build/Linux/RelWithDebInfo/ && ./onnxruntime_test_all --gtest_filter=GradientCheckerTest.CosGrad`

### CODE REVIEW

File: orttraining/orttraining/core/graph/gradient_builder_base.h [hunk 1]
```diff
@@ -295,6 +295,10 @@ class GradientBuilderBase {
     return ConstantScalarNode(1.0f, "OneConstant_Type" + std::to_string(elem_type), elem_type);
   }
 
+  static NodeDef MinusOneConstantNode(int elem_type) {
```
Comment (iK1D) [inline orttraining/orttraining/core/graph/gradient_builder_base.h:298]: We can do (0 - Sin(I(0)) * GO(0) so we don't need this MinusOneConstant (we already have ZeroConstant), and maybe from And/Sub is faster than Mul/Div in computing.

**Review by iK1D — COMMENTED** [review]
Since we don't have CUDA kernel for SinGrad, I suggest also change the SinGrad to using Ops to compose.

File: orttraining/orttraining/core/graph/gradient_builder_base.h (hunk 1 above)
Comment (prathikr) [reply orttraining/orttraining/core/graph/gradient_builder_base.h]: Done!

**Comment by prathikr** [comment]
@iK1D I've implemented the singrad change in a separate PR: https://github.com/microsoft/onnxruntime/pull/13263

**Comment by prathikr** [comment]
> Since we don't have CUDA kernel for SinGrad, I suggest also change the SinGrad to using Ops to compose.

@iK1D I've implemented the singrad change in a separate PR: https://github.com/microsoft/onnxruntime/pull/13263

**Review by baijumeswani — COMMENTED** [review]
Can we add an ortmodule python test to make sure pytorch model having cos can run well with ORTModule?

**Review by iK1D — APPROVED** [review]

**Review by askhade — APPROVED** [review]
LGTM

**Pull request merged**